src/awsexplorer/awsExplorer.ts

```diff
@@ -100,4 +92,32 @@ export class AwsExplorer implements vscode.TreeDataProvider<AWSTreeNodeBase>, Re
     public refresh(node?: AWSTreeNodeBase) {
         this._onDidChangeTreeData.fire(node)
     }
+
+    private async getRootNodes(): Promise<AWSTreeNodeBase[]> {
+        if (!(await this.awsContext.getCredentials())) {
+            return [ROOT_NODE_SIGN_IN]
+        }
+
+        const userVisibleRegionCodes = await this.awsContext.getExplorerRegions()
+        const regionMap = toMap(await this.regionProvider.getRegionData(), r => r.regionCode)
+
+        return await makeChildrenNodes({
+            getChildNodes: async () => {
+                updateInPlace(
+                    this.regionNodes,
+                    intersection(regionMap.keys(), userVisibleRegionCodes),
+                    key => this.regionNodes.get(key)!.update(regionMap.get(key)!),
+                    key => new RegionNode(regionMap.get(key)!, this.regionProvider)
+                )
+
+                return [...this.regionNodes.values()]
```

Comment:
Why are we returning a copy? is this stored anywhere?

Reply:
`.values()` is an iterator, this is simply converting it to an array. Is there a recommended way to do this? This is done throughout the codebase, but I'd want to stop doing this wrong.